python-checks/src/main/java/org/sonar/python/checks/BuiltinShadowingAssignmentCheck.java

```diff
@@ -86,15 +93,35 @@ private void checkAnnotatedAssignment(SubscriptionContext ctx) {
   }
 
   private void raiseIssueForNonGlobalVariable(SubscriptionContext ctx, Name variable) {
-    Symbol variableSymbol = variable.symbol();
-    if (variableSymbol != null && variableSymbol.usages().stream().noneMatch(u -> u.kind().equals(Usage.Kind.GLOBAL_DECLARATION))) {
-      PreciseIssue existingIssue = variableIssuesRaised.get(variableSymbol);
-      if (existingIssue != null) {
-        existingIssue.secondary(variable, REPEATED_VAR_MESSAGE);
-      } else {
-        variableIssuesRaised.put(variableSymbol, ctx.addIssue(variable, MESSAGE));
-      }
-    }
+    Optional.ofNullable(variable.symbol())
+      .filter(symbol -> symbol.usages().stream().map(Usage::kind).noneMatch(Usage.Kind.GLOBAL_DECLARATION::equals))
+      .ifPresent(symbol -> {
+        var existingIssue = variableIssuesRaised.get(symbol);
+        if (existingIssue != null) {
+          existingIssue.secondary(variable, REPEATED_VAR_MESSAGE);
+        } else {
+          var issue = (IssueWithQuickFix) ctx.addIssue(variable, MESSAGE);
+
+          PythonQuickFix quickFix = createQuickFix(symbol);
+          issue.addQuickFix(quickFix);
+          variableIssuesRaised.put(symbol, issue);
+        }
+      });
+  }
+
+  private static PythonQuickFix createQuickFix(Symbol symbol) {
+    var edits = symbol.usages()
+      .stream()
+      .map(Usage::tree)
+      .map(Tree::firstToken)
+      .sorted(Comparator.comparing(Token::line).reversed().thenComparing(Token::column).reversed())
```

Comment:
Why use the `reversed` method? (it's used twice on this line)
Do we need to sort? (maybe we do)

Reply:
removed, I did several tests including calling quick fix in VS code, and didn't find that it makes sense. Good catch